x/mint/types/params.go

```diff
@@ -1,24 +1,183 @@
+		paramtypes.NewParamSetPair(KeyInflationChange, &p.InflationChange, validateInflationChange),
+		paramtypes.NewParamSetPair(KeyMaxBlockDuration, &p.MaxBlockDuration, validateMaxBlockDuration),
+		paramtypes.NewParamSetPair(KeyInflationRecipients, &p.InflationRecipients, validateInflationRecipients),
+	}
+}
+
+// String implements the Stringer interface.
+func (p Params) String() string {
+	out, _ := yaml.Marshal(p)
+	return string(out)
 }
 
 // Validate perform object fields validation.
-func (m Params) Validate() error {
-	panic("unimplemented 👻")
+func (p Params) Validate() error {
+	if err := validateInflation(p.MinInflation); err != nil {
+		return sdkErrors.Wrap(err, "min_inflation param has invalid value, should be between 0 and 1")
+	}
+	if err := validateInflation(p.MaxInflation); err != nil {
+		return sdkErrors.Wrap(err, "max_inflation param has invalid value, should be between 0 and 1")
+	}
+	if err := validateBondedRatio(p.MinBonded); err != nil {
+		return sdkErrors.Wrap(err, "min_bonded param has invalid value, should be between 0 and 1")
+	}
+	if err := validateBondedRatio(p.MaxBonded); err != nil {
+		return sdkErrors.Wrap(err, "max_bonded param has invalid value, should be between 0 and 1")
+	}
+	if err := validateInflationChange(p.InflationChange); err != nil {
+		return sdkErrors.Wrap(err, "inflation_change param has invalid value, should be between 0 and 1")
+	}
+	if err := validateInflationRecipients(p.InflationRecipients); err != nil {
+		return err
+	}
+	return nil
+}
+
+func validateInflation(i interface{}) error {
+	inflation, ok := i.(sdk.Dec)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+	if !isDecValidPercentage(inflation) {
+		return ErrInvalidInflation
+	}
+	return nil
+}
+
+func validateBondedRatio(i interface{}) error {
+	bondedRatio, ok := i.(sdk.Dec)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+	if !isDecValidPercentage(bondedRatio) {
+		return ErrInvalidBondedRatio
+	}
+	return nil
+}
+
+func validateInflationChange(i interface{}) error {
+	inflationChange, ok := i.(sdk.Dec)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+	if !isDecValidPercentage(inflationChange) {
+		return ErrInvalidInflationChange
+	}
+	return nil
+}
+
+func validateMaxBlockDuration(i interface{}) error {
+	v, ok := i.(int64)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+
+	if v < 0 {
+		return fmt.Errorf("max block duration must be non-negative")
+	}
+
+	return nil
+}
+
+func validateInflationRecipients(i interface{}) error {
+	inflationRecipients, ok := i.([]*InflationRecipient)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+	if len(inflationRecipients) < 1 {
+		return sdkErrors.Wrap(ErrInvalidInflationRecipient, "inflation recipients not found")
+	}
+	inflationDistribution := sdk.ZeroDec()
+	for _, recipient := range inflationRecipients {
+		inflationDistribution = inflationDistribution.Add(recipient.Ratio)
```

Comment:
we should add a check here that the inflation recipient is valid, aka not empty.